homeassistant/components/skybell/config_flow.py

```diff
@@ -17,6 +18,39 @@
 class SkybellFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Skybell."""
 
+    reauth_email: str
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle a reauthorization flow request."""
+        self.reauth_email = entry_data[CONF_EMAIL]
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle user's reauth credentials."""
+        errors = {}
+        if user_input:
+            password = user_input[CONF_PASSWORD]
+            entry_id = self.context["entry_id"]
```

Comment:
I recall there was a problem with the `self.context` value disappearing between steps if they re-entered the step after error.  I'm not 100% sure if that is still an issue. Its likely safer to save the entry in `async_step_reauth` as a bound var.

I think we have something like this in other integrations to make sure its not an issue.

```
    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
        """Handle a reauthorization flow request."""
        self.reauth_email = entry_data[CONF_EMAIL]
        self.reauth_entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
        return await self.async_step_reauth_confirm()

```

Someone might know if the above is fixed but I can't tell from a brief glance through the code.

Reply:
Maybe but I just tested the flow with an incorrect password and then the correct one. It is working as expected. Email is in the `entry.data` as expected, unchanged.